Lib/pathlib.py

```diff
@@ -300,18 +300,24 @@ def __reduce__(self):
         return (self.__class__, self.parts)
 
     def __init__(self, *args):
-        if not args:
-            path = ''
-        elif len(args) == 1:
-            path = os.fspath(args[0])
+        paths = []
+        for arg in args:
+            if isinstance(arg, PurePath):
+                path = arg._raw_path
+            else:
+                path = os.fspath(arg)
+                if not isinstance(path, str):
+                    raise TypeError(
+                        "argument should be a str or an os.PathLike "
+                        "object where __fspath__ returns a str, "
+                        f"not {type(path).__name__!r}")
```

Comment:
Give improved message for any bad argument.  See edited comment on issue.
```suggestion
                try:
                    path = os.fspath(arg)
                except TypeError:
                    path = arg
                if not isinstance(path, str):
                    raise TypeError(
                        "argument should be a str or an os.PathLike "
                        "object where __fspath__ returns a str, "
                        f"not {type(path).__name__!r}")
```

Reply:
I didn't see this solution. Very nice :)

Reply:
I think it should be "an str" following this rule: https://www.btb.termiumplus.gc.ca/tpv2guides/guides/wrtps/index-eng.html?lang=eng&lettr=indx_catlog_a&page=9MN1DTkGtnPE.html#:~:text=When%20the%20indefinite%20article%20is,before%20the%20consonant%20sound%20n)

Even though `str` is not an acronym, it's pronounced as an acronym, i.e. "as tee arr". I don't know if there was a discussion about this but there's a couple places in the docs where "a str" is used:
library/io.rst:      Return a ``str`` containing the entire contents of the buffer.
library/importlib.resources.rst:    Read and return the contents of *resource* within *package* as a ``str``.

And one place were "an str" is used.

Reply:
'an str' looks completely wrong to me.  'str' is an abbreviation for 'string' and is neither an acronym or initialism, the subject of the reference.  I read it as 'string' or possibly 'stir', not as 'es tee are'.  I would change the occurrence of 'an str'.